ebos/equil/initstateequil.hh

```diff
@@ -945,17 +945,29 @@ public:
           rv_(grid.size(/*codim=*/0))
     {
         //Check for presence of kw SWATINIT
-        if (eclipseState.get3DProperties().hasDeckDoubleGridProperty("SWATINIT") && applySwatInit) {
-            const std::vector<double>& swatInitEcl = eclipseState.
-                get3DProperties().getDoubleGridProperty("SWATINIT").getData();
+        if (applySwatInit) {
             const int nc = grid.size(/*codim=*/0);
             swatInit_.resize(nc);
-            const int* gc = Opm::UgGridHelpers::globalCell(grid);
-            for (int c = 0; c < nc; ++c) {
-                const int deckPos = (gc == NULL) ? c : gc[c];
-                swatInit_[c] = swatInitEcl[deckPos];
+
+#ifdef ENABLE_3DPROPS_TESTING
+            if (eclipseState.fieldProps().has<double>("SWATINIT")) {
+                const std::vector<double>& swatInitEcl = eclipseState.fieldProps().get<double>("SWATINIT");
+                for (int c = 0; c < nc; ++c)
```

Comment:
Is there a special reason for manual copying instead of
`swatInit_ = eclipseState.fieldProps().get<double>("SWATINIT");

Reply:
A guiding principle for these changes have to a large extent been to minimize the changes; at least until things get through and "work" - but this one was maybe exaggerating it. I have updated to just do `swatInit_ = ...`.